### PR TITLE
Associate key value device person

### PR DESCRIPTION
**Pull request by aurora-labs**
https://trello.com/c/t1qMs5Ko

### CODE REVIEW

File: sdk/src/main/java/com/tendarts/sdk/TendartsSDK.java [hunk 1]
```diff
@@ -1593,6 +1593,66 @@ public void onFail(int operationId, String reason, Communications.PendingCommuni
...
+	/**
+	 * Associates Key-Value with user
+	 * @param context
+	 * @param key
+	 * @param kind
+	 * @param value
+	 * @param observer
+	 */
+	public static void associateKeyValueWithUser(Context context,
+												   String key, TDKeysHandler.KeyValueKind kind, String value,
```
Comment (awayo) [inline sdk/src/main/java/com/tendarts/sdk/TendartsSDK.java]: wrong indentation

File: sdk/src/main/java/com/tendarts/sdk/client/TDKeysHandler.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+            this.kindValue = kindValue;
+        }
+
+        public int getKindValue() {
+            return kindValue;
+        }
+
+    }
+
+    public static void associateKeyValueWithDevice(final Context context,
```
Comment (awayo) [inline sdk/src/main/java/com/tendarts/sdk/client/TDKeysHandler.java]: For an API if we want Kotlin interoperability add @NonNull or @Nullable must be necessary.

File: sdk/src/main/java/com/tendarts/sdk/client/TDKeysHandler.java [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+                listener.onError("The user should be already registered");
+            }
+            return;
+        }
+
+
+        String data = null;
+
+        try {
+            JSONObject keyJson = new JSONObject();
```
Comment (awayo) [inline sdk/src/main/java/com/tendarts/sdk/client/TDKeysHandler.java:148]: Can't this be refactorized ? i see this code repeated again and again

**Review by awayo — CHANGES_REQUESTED** [review]
I left some comments.

File: sdk/src/main/java/com/tendarts/sdk/client/TDKeysHandler.java (hunk 3 above)
Comment (aurora-labs) [reply sdk/src/main/java/com/tendarts/sdk/client/TDKeysHandler.java:148]: It's two times in this class, but with different values, do you refer to that?

**Pull request merged**